R/calculate-effect-sizes.R

```diff
@@ -410,6 +411,35 @@ batch_calc_ES <- function(dat,
   
   ES_long_names <- names(ES_ests_long)
   
+  # weights
+  conditions <- dat %>% dplyr::select({{condition}}) %>% unique() %>% dplyr::pull()
+  
+  if (is.null(baseline_phase)) baseline_phase <- conditions[1]
+  
+  if (is.null(intervention_phase)) {
+    intervention_phase <- setdiff(conditions, baseline_phase)[1]
+    if (length(conditions) > 2) {
+      msg <- paste0("The 'condition' variable has more than two unique values. Treating '", intervention_phase, "' as the intervention phase.")
+      warning(msg)
+    }
+  }
+  
+  dat <- 
+    dat %>% 
+    dplyr::filter(!!rlang::sym(condition) %in% c(baseline_phase, intervention_phase)) %>% 
+    dplyr::mutate(!!rlang::sym(condition) := ifelse(!!rlang::sym(condition) == baseline_phase, "A", "B"))
+  
+  n_weights <- 
+    dat %>% 
+    dplyr::group_by(!!!rlang::syms(c(grouping, aggregate, condition))) %>% 
+    dplyr::summarise(n = dplyr::n()) %>% 
+    tidyr::pivot_wider(names_from = !!rlang::sym(condition), values_from = n) %>% 
+    dplyr::mutate(
+      AB = A * B,
+      sum_inv = 1/A + 1/B

```

Comment:
Could you calculate the appropriate weight in this step? Maybe use `case_when()`? This would probably make the big long set of conditionals (lines 457-492) much simpler.